test/fasta/fasta_test.cpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <fstream>
+#include <string>
+
+#include "catch/catch.hpp"
+
+#include "fasta/fasta.hpp"
+
+namespace ebi
+{
+  const std::string WHOLE_GENOME_FASTA = "test/input_files/fasta/whole_genome.fa";
+  const std::string WHOLE_GENOME_FASTA_INDEX = "test/input_files/fasta/whole_genome.fa.fai";
+  const std::string SINGLE_CHROMOSOME_FASTA = "test/input_files/fasta/single_chromosome.fa";
+  const std::string SINGLE_CHROMOSOME_FASTA_INDEX = "test/input_files/fasta/single_chromosome.fa.fai";
+
+  TEST_CASE("Tests for using file based fasta", "[fasta]")
+  {
+      SECTION("Create file based fasta objects")
+      {
+          CHECK_NOTHROW(std::unique_ptr<ebi::vcf::fasta::IFasta>(
+                  new ebi::vcf::fasta::FileBasedFasta(WHOLE_GENOME_FASTA, WHOLE_GENOME_FASTA_INDEX)));
+          CHECK_NOTHROW(std::unique_ptr<ebi::vcf::fasta::IFasta>(
+                  new ebi::vcf::fasta::FileBasedFasta(SINGLE_CHROMOSOME_FASTA, SINGLE_CHROMOSOME_FASTA_INDEX)));
+      }
+
+      SECTION("Check if contigs exist, whole genome")
+      {
+          std::unique_ptr<ebi::vcf::fasta::IFasta> fasta(
+                  new ebi::vcf::fasta::FileBasedFasta(WHOLE_GENOME_FASTA, WHOLE_GENOME_FASTA_INDEX));
+
+          CHECK(fasta->count("I") == 1);
+          CHECK(fasta->count("II") == 1);
+          CHECK(fasta->count("III") == 1);
+          CHECK(fasta->count("Unexisted") == 0);
```

Comment:
Not critical, but for language correctness:

```suggestion
          CHECK(fasta->count("Non-existing") == 0);
```

Applies to others below.